cogs/tibia.py

```diff
@@ -221,6 +225,15 @@ def __init__(self, bot: NabBot):
         except CannotPaginate as e:
             await ctx.send(e)
 
+    @staticmethod
+    def _get_cached_user_(self, user_id, user_unknown, users_cache, user_servers):
+        cached_user = users_cache.get(user_id)
+        if cached_user is None:
+            member_user = self.bot.get_member(user_id, user_servers)
+            cached_user = user_unknown if member_user is None else member_user
+            users_cache[user_id] = cached_user
+        return cached_user
```

Comment:
I think it would be better to have this return `Union[User, None]` instead of `Union[str, User]` so the possible values are similar, and in the implementations you don't have to deal with having a variable for "user_unknown" to compare on later.

Instead of using `Dict.get()`, you can use `key in Dict` to check if the user is in the cache, if it's not, then fetch.

I know I have committed this sin in the past, but I'm working on it!